internal/reader/contractreader/contractreader_facade.go

```diff
@@ -18,4 +18,9 @@ type ContractReaderFacade interface {
 	Unbind(ctx context.Context, bindings []types.BoundContract) error
 	QueryKey(ctx context.Context, contract types.BoundContract, filter query.KeyFilter, limitAndSort query.LimitAndSort, sequenceDataType any) ([]types.Sequence, error)
 	//mustEmbedUnimplementedContractReaderServer()
+	Name() string
```

Comment:
This make contractreader_facade to implement ContractReader interface

Reply:
Why do we want that? You could also accomplish that by embedding `service.Service`, [which is how the ContractReader does it](https://github.com/smartcontractkit/chainlink-common/blob/d4cf7aff85cb3f2a0630bc2a11b42cfcbb80c40c/pkg/types/contract_reader.go#L22).

I thought we left out `Unbind` to narrow the interface even.

Reply:
Oh, nice it's gonna be cleaner to just embed `services.Service`

I've added it to do something like this in tests
```go
	emptyReader := reader.NewMockContractReaderFacade(t)
	faultyReader := reader.NewMockContractReaderFacade(t)
	validReader := reader.NewMockContractReaderFacade(t)


	contactReaders := map[cciptypes.ChainSelector]types.ContractReader{
		faultyChain: faultyReader,
		emptyChain:  emptyReader,
		validChain:  validReader,
	}
```